src/compiler/program.ts

```diff
@@ -609,11 +615,42 @@ namespace ts {
             }
         }
 
+        const outputFingerprints: Map<OutputFingerprint> =
+            options.watch && sys.createHash && sys.getModifiedTime ? {} : undefined;
+
+        const fileWriter: typeof sys.writeFile = outputFingerprints ?

```

Comment:
@mhegazy I personally do not like parallel branches with same condition (because they can not indicate a strong relationship between those branches), so if I write them into a single function, I would like a single `if...else` with two `sys.writeFile` in each branch. And after merging them together, function `fileWriter` would be awkward to exist alone with `writeFile`. If I inline `fileWriter` into `writeFile`, it would result in a long `try` block. :cry: What's your suggestion?